src/scripts/main.js

```diff
@@ -1,3 +1,37 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('thead th');
+
+  if (!target) {
+    return;
+  }
+  sort(target);
+});
+
+function sort(elem) {
+  // we find the column index number
+  const thead = [...table.rows[0].cells];
+  const currentPos = thead.indexOf(elem);
+
+  const tableBody = [...table.rows].slice(1, -1).sort((a, b) => {
```

Comment:
The slice method is used incorrectly here. It should include the last row of the table for sorting. Use slice(1) instead of slice(1, -1) to include all rows except the header.